doc/modules/neighbors.rst

```diff
@@ -427,6 +482,35 @@ and the ``'effective_metric_'`` is in the ``'VALID_METRICS'`` list of
 same order as the number of training points, and that ``leaf_size`` is 
 close to its default value of ``30``.
 
+Valid Metrics for Nearest Neighbor Algorithms
+---------------------------------------------
+
+========================     =================================================================
+Algorithm                    Valid Metrics
+========================     =================================================================
+**Brute Force**              'euclidean', 'l2', 'l1', 'manhattan', 'cityblock',
+                             'braycurtis', 'canberra', 'chebyshev', 'correlation',
+                             'cosine', 'dice', 'hamming', 'jaccard', 'kulsinski',
+                             'mahalanobis', 'matching', 'minkowski', 'rogerstanimoto',
+                             'russellrao', 'seuclidean', 'sokalmichener',
+                             'sokalsneath', 'sqeuclidean', 'yule', 'wminkowski'
+
+**K-D Tree**                 'chebyshev', 'euclidean', 'cityblock', 'manhattan', 'infinity',
+                             'minkowski', 'p', 'l2', 'l1'
+
+**Ball Tree**                'chebyshev', 'sokalmichener', 'canberra', 'haversine',
+                             'rogerstanimoto', 'matching', 'dice', 'euclidean', 'braycurtis',
+                             'russellrao', 'cityblock', 'manhattan', 'infinity', 'jaccard',
+                             'seuclidean', 'sokalsneath', 'kulsinski', 'minkowski',
+                             'mahalanobis', 'p', 'l2', 'hamming', 'l1', 'wminkowski', 'pyfunc'
+========================     =================================================================
+
+A list of valid metrics for the above algorithms can be obtained by:
+
+    >>> from sklearn.neighbors import KDTree
+    >>> print(KDTree.valid_metrics)           # doctest: +ELLIPSIS
+    [...]
```

Comment:
Had to `...` this as print was giving different output everytime :| Not sure how it will fail if someone modifies the valid metrics.